packages/puppeteer-parse/index.js

```diff
@@ -397,15 +397,33 @@ function validateUrlString(url) {
   }
 }
 
+function tryParseUrl(urlStr) {
+  if (!urlStr) {
+    return null;
+  }
+  
+  // a regular expression to match URLs
+  var regex = /(https?:\/\/[^\s]+)/g;
```

Comment:
We might need to remove `g` in the regex as it won't include capture groups in matches

Reply:
Thanks for comment, I prefer to keep `g` here, see different regex expr with different results below: 

![image](https://user-images.githubusercontent.com/8675075/223333549-014a4988-f550-43ea-8026-b68f40ff0600.png)

I think if we need any change in the future, this expression is less likely to be updated.

Reply:
`var` is also changed to `const`

Reply:
Yeah, good point!